### PR TITLE
Retry upon connection error

### PR DESCRIPTION
**Pull request by tzuhsial**
Modifications I added to process 5000 documents for 4~5 hours

PS: I believe you will need to add python2 compatibility before packaging it to pypi....

### CODE REVIEW

**Comment by tzuhsial** [comment]
@yuwen41200 Please check

File: ckipclient/ckipclient.py [hunk 1]
```diff
@@ -37,9 +37,20 @@ def __init__(self, ip: str, port: int, usr: str, pwd: str, safe: bool = True):
...
 
-    def segment(self, text: str, pos: bool = True) -> list:
+    def segment(self, text: str, pos: bool = True):
+        status = False
+        results = None
+        while not status:
+            try:
+                results = self._segment(text, pos)
+                status = True
+            except ConnectionError:
```
Comment (yuwen41200) [inline ckipclient/ckipclient.py]: How about checking other errors as well?
For example:
```python
def safe_segment(message):
    while True:
        try:
            message = segment(message) if message else None
        except ElementTree.ParseError as err:
            print('ElementTree.ParseError:', err, flush=True)
            print('wait 10 seconds...', flush=True)
            time.sleep(10)
        except ConnectionError as err:
            print('ConnectionError:', err, flush=True)
            print('wait 10 seconds...', flush=True)
            time.sleep(10)
        except OSError as err:
            print('OSError:', err, flush=True)
            print('wait 10 seconds...', flush=True)
            time.sleep(10)
        else:
            return message
```

File: ckipclient/ckipclient.py [hunk 2]
```diff
@@ -37,9 +37,20 @@ def __init__(self, ip: str, port: int, usr: str, pwd: str, safe: bool = True):
         self.usr = usr
         self.pwd = pwd
         self.safe = safe
-        self.counter = 0
 
-    def segment(self, text: str, pos: bool = True) -> list:
+    def segment(self, text: str, pos: bool = True):
```
Comment (yuwen41200) [inline ckipclient/ckipclient.py]: For consistency, please keep the type hints.

**Review by yuwen41200 — COMMENTED** [review]
Thanks! Please see my comments.

**Comment by yuwen41200** [comment]
@iammrhelo Regarding Python 2 compatibility, sorry but I have no plan to add Python 2 support.

File: ckipclient/ckipclient.py (hunk 2 above)
Comment (tzuhsial) [reply ckipclient/ckipclient.py]: Oops...accidentally modified that part

File: ckipclient/ckipclient.py (hunk 1 above)
Comment (tzuhsial) [reply ckipclient/ckipclient.py]: Changed function name to 'safe_segment'
Since the parsing code is yours, I'll leave the rest of the errors to you.

**Comment by tzuhsial** [comment]
@yuwen41200 
If it doesn't support Python2, 
I think you should add that into the Readme.(and not package it to pypi)

Anyways, for this commit
1. added missing type hint
2. changed function name for consistency
   - 'segment' -> 'safe_segment'
   - '_segment' -> 'segment'
3. add 'retry' parameter in 'safe_segment' to specify number of retries
Please check, thanks!

**Comment by tzuhsial** [comment]
@yuwen41200 Forgot to mention, thanks a lot for your work!

**Review by yuwen41200 — APPROVED** [review]
Thanks.

**Comment by yuwen41200** [comment]
@iammrhelo 
OK, I added a notice in README.
But... I should not upload it to PyPI?! Why?
As far as I know, supporting Python 2 is NOT a must.
I've already specified its Python version correctly, conforming to the requirements as stated in:
https://packaging.python.org/tutorials/distributing-packages/

**Pull request merged**